website/docs/r/kubernetes_cluster.html.markdown

```diff
@@ -448,6 +450,19 @@ An `identity` block supports the following:
 
 ---
 
+A `key_vault_kms` block supports the following:
+
+* `enabled` - (Optional) Whether to enable Azure Key Vault key management service. The default is `false`.
+
+* `key_id` - (Optional) Identifier of Azure Key Vault key. See [key identifier format](https://learn.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name) for more details. When Azure Key Vault key management service is enabled, this field is required and must be a valid key identifier. When `enabled` is `false`, leave the field empty.
+
+* `key_vault_network_access` - (Optional) Network access of the key vault
+Network access of key vault. The possible values are `Public` and `Private`. `Public` means the key vault allows public access from all networks. `Private` means the key vault disables public access and enables private link. The default value is `Public`.
+
+* `key_vault_resource_id` - (Optional) Resource ID of key vault. When `key_vault_network_access` is `Private`, this field is required and must be a valid resource ID. When `key_vault_network_access` is `Public`, leave the field empty.
```

Comment:
I would recommend adding the relevant check in aks resource's `expandKubernetesClusterAzureKeyVaultKms` function.

Reply:
Will do

Reply:
done